Add API-M Tutorial scenarios (#3715)
wso2/docs-apim: en/docs/tutorials/scenarios/scenario2-access-control.md

> # Scenario 2 - Engage Access control to the API

Reviewer: ```suggestion
# Scenario 2 - Engage Access Control to the API
```

Author: Fixed

## User Story

**ColTrain has a separate API to manage schedules for their internal staff. This API needs to have more elevated permission levels to access than their public API. All the employees in the ColTrain company have access to the end-user application where they can view the train schedule details using this API.  All the staff in the ColTrain should be able to check the available schedules whereas only the staff with admin privileges can add, edit or remove the existing schedule. Any other registered or public user should not be able to view this API since it is there for internal tasks. Coltrain wants to have a clear separation on who can view and access their APIs. They have identified that it would be a cumbersome task If they are to implement this from scratch to their backend APIs directly. Since now they are using an API Management platform, they wanted to move all these authentication and authorization tasks out of their internal APIs. This would be beneficial for their internal teams because they only have to pay attention to their APIs business logic only.** 

**_Time to Complete : 7 mins_**

> We could configure the API to be visible for a set of users. For example, This API should be visible for only dev portal users with **coltrain_employee** role only.

Reviewer: ```suggestion
We could configure the API to be visible for a set of users. For example, this API should be visible for only Developer Portal users with **coltrain_employee** role only.
```

Author: Fixed

![Throttling description]({{base_path}}/assets/img/tutorials/scenarios/throttling-description.png)

> Also WSO2 API manager provides capability to provide access control to the resources of the API by using oauth2 scopes. Requests containing access tokens with the correct scope will be able to access these resources. 

Reviewer: ```suggestion
Also WSO2 API Manager provides capability to provide access control to the resources of the API by using OAuth2 scopes. Requests containing access tokens with the correct scope will be able to access these resources. 
```

Author: Fixed

![Access control description]({{base_path}}/assets/img/tutorials/scenarios/acl-description.png)

> Note: This setup contains roles **schedule_admin**  and **coltrain_employee** already created in the ColTrain tenant domain. **schedule_admin** and **coltrain_employee** roles are assigned to the user **jenny@coltrain.com** and only **coltrain_employee** role is assigned to **george@coltrain.com** . 

Reviewer: ```suggestion
!!! Note 
    This setup contains roles **schedule_admin**  and **coltrain_employee** already created in the ColTrain tenant domain. **schedule_admin** and **coltrain_employee** roles are assigned to the user **jenny@coltrain.com** and only **coltrain_employee** role is assigned to **george@coltrain.com** . 
```

Author: Fixed

Lets create a separate API named **ColTrainInternalTimeTableAPI** for this and set the visibility of that API based on roles. 

> 1. Log in to the publisher portal again `https://localhost:9443/publisher/` . Use user as _apiprovider@coltrain.com_ and password as _user123_

Reviewer: ```suggestion
1. Log on to the Publisher portal again `https://localhost:9443/publisher/` . Use user credentials as _apiprovider@coltrain.com_ and password as _user123_.
```

Author: fixed

2. Create a new API using the OpenAPI definition **coltrain-openapi.yaml** provided in the _resources_ location. Lets use _/coltrain-schedule_ as the context. Use the endpoint provided in the file as it is.

    ![Create API]({{base_path}}/assets/img/tutorials/scenarios/coltrain-internal-api-create.png)

3. Go to **Develop → Portal Configurations → Basic Info** and under the **Developer portal visibility**, select **Restrict by role(s)** . This would enable another field to set the role. Use **coltrain_employee** and press enter and save.

    ![Set visibility]({{base_path}}/assets/img/tutorials/scenarios/set-visibility.png)

> Developer portal visibility is set to the API. Users with **coltrain_employee** roles can now view the API in the developer portal. 

Reviewer: ```suggestion
Developer Portal visibility is set to the API. Users with the **coltrain_employee** role can now view the API in the Developer Portal. 
```

Author: fixed
